code/game/machinery/computer/ai_core.dm

```diff
@@ -4,15 +4,20 @@
 	name = "AI core"
 	icon = 'icons/mob/AI.dmi'
 	icon_state = "0"
+	obj_integrity = 500
+	max_integrity = 500
 	var/state = 0
 	var/datum/ai_laws/laws = null
 	var/obj/item/circuitboard/circuit = null
 	var/obj/item/mmi/brain = null
 
 /obj/structure/AIcore/Destroy()
-	QDEL_NULL(laws)
-	QDEL_NULL(circuit)
-	QDEL_NULL(brain)
+	if(circuit)
+		qdel(circuit)
+		circuit = null
+	if(brain)
+		brain.forceMove(loc)
+		brain = null
```

Comment:
what existed before was superior. The brain change is a balance change though.

Reply:
reverted it